src/db.rs

```diff
@@ -94,26 +94,32 @@ impl DB {
             let split_line = line.rsplitn(3, '|').collect::<Vec<&str>>();
 
             match split_line.as_slice() {
-                [_, rank_str, path_str] => {
-                    // Set last_accessed to current time; otherwise, the rank will get scaled down,
-                    // depending on how old the imported entry is (import in-place and let zoxide
-                    // scale down organically)
-                    let epoch = util::get_current_time()?;
+                [epoch_str, rank_str, path_str] => {
+                    let epoch = epoch_str
+                        .parse::<i64>()
+                        .with_context(|| anyhow!("could not parse epoch: '{}'", epoch_str))?;
                     let rank = rank_str
                         .parse::<f64>()
                         .with_context(|| anyhow!("could not parse rank: '{}'", rank_str))?;
+                    let path = match PathBuf::from(path_str).canonicalize() {
+                        Ok(path) => path,
+                        Err(e) => {
+                            eprintln!("invalid path '{}' at line {}: {}", path_str, line_number, e);
+                            continue;
+                        }
+                    };
 
                     // FIXME: When we switch to PathBuf for storing directories inside Dir, just
                     // pass `PathBuf::from(path_str)`
                     self.dirs.push(Dir {
-                        path: path_str.to_string(),
+                        path: path.display().to_string(),
```

Comment:
AFAIK, this is a lossy conversion, and would create an invalid database entry. Use `to_str()` instead, and warn the user if the path contains invalid Unicode.

Reply:
You are indeed correct -- I always forget about this ^^;